apps/server/lib/lexical/server/transport/std_io.ex

```diff
@@ -112,10 +112,16 @@ defmodule Lexical.Server.Transport.StdIO do
     end
   end
 
-  defp read(device, amount) do
+  defp read_body(device, amount) do
     case IO.read(device, amount) do
-      data when is_binary(data) or is_list(data) -> {:ok, data}
-      other -> other
+      data when is_binary(data) or is_list(data) ->
+        # This is a bit "magical" and is likely a symptom of a bug elsewhere.
```

Comment:
I take slight issue with this comment. This is the lowest level of lexical. If the bug is elsewhere, where is it? You established that the binaries decoded here were not
Utf8.  Where else would the bug be?

Reply:
"Bug" may be the wrong word -- "misunderstanding" is probably more accurate.

How about:

```elixir
# Ensure that incoming data is latin1 to prevent double-encoding to utf8 later
```

Reply:
```suggestion
        # Ensure that incoming data is latin1 to prevent double-encoding to utf8 later
```